maestrowf/interfaces/script/slurmscriptadapter.py

```diff
@@ -116,22 +117,49 @@ def get_header(self, step):
         :returns: A string of the header based on internal batch parameters and
             the parameter step.
         """
-        resources = ChainMap(step.run, self._batch)
+
+        resources = {}
+        resources.update(step.run)
```

Comment:
This produces incorrect behavior because the update is backwards here -- in this order, the `batch` settings will override the more accurate and specific step settings. I just tested it, and this is in fact the case. I would do something like:

```
resources = {}
resources.update(self._batch)
resources.update({ key : value for key, value in step.run if value })
```

From there, you can check `procs` and `nodes` and throw the exception if both aren't set, run through the resources, and then add `ntasks` if needed.

Reply:
Sorry, I think I got the dictionary comprehension wrong last night -- it should look something like:
```
resources.update({ key : value for (key, value) in step.run if value })
```